src/gui_common/EditorEntryButton.tscn

```diff
@@ -0,0 +1,196 @@
+[gd_scene load_steps=11 format=3 uid="uid://jjb8ei8wuuk"]
+
+[ext_resource type="Texture2D" uid="uid://c732snfs7i3xx" path="res://assets/textures/gui/bevel/LargeTwoNormal.png" id="1_ixss3"]
+[ext_resource type="Texture2D" uid="uid://bd4c6guswatde" path="res://assets/textures/gui/bevel/LargeTwoPressed.png" id="2_suams"]
+[ext_resource type="Texture2D" uid="uid://cbc1ld512pl71" path="res://assets/textures/gui/bevel/LargeTwoHover.png" id="3_kbqhu"]
+[ext_resource type="Texture2D" uid="uid://dcc6nq6ig4etg" path="res://assets/textures/gui/bevel/LargeTwoDisabled.png" id="4_4niqg"]
+[ext_resource type="Texture2D" uid="uid://cai50f52nr8gw" path="res://assets/textures/background_particle_fuzzy.png" id="5_vcgwd"]
+[ext_resource type="Texture2D" uid="uid://dagkk6puhnfvu" path="res://assets/textures/gui/bevel/ReproductionBar.png" id="6_nj585"]
+[ext_resource type="Texture2D" uid="uid://bf55j5u0k0oc1" path="res://assets/textures/gui/bevel/AmmoniaInv.png" id="7_wu333"]
+[ext_resource type="Texture2D" uid="uid://bp6urdgi1p1hq" path="res://assets/textures/gui/bevel/PhosphateInv.png" id="8_yin80"]
+
+[sub_resource type="Animation" id="30"]
+length = 5.0
+loop_mode = 1
+tracks/0/type = "value"
+tracks/0/imported = false
+tracks/0/enabled = true
+tracks/0/path = NodePath("Highlight:modulate")
+tracks/0/interp = 1
+tracks/0/loop_wrap = true
+tracks/0/keys = {
+"times": PackedFloat32Array(0, 1, 2.5, 3.5, 5),
+"transitions": PackedFloat32Array(0.5, 0.5, 0.5, 1, 1),
+"update": 0,
+"values": [Color(1, 1, 1, 0), Color(1, 1, 1, 0), Color(1, 1, 1, 1), Color(1, 1, 1, 1), Color(1, 1, 1, 0)]
+}
+tracks/1/type = "value"
+tracks/1/imported = false
+tracks/1/enabled = true
+tracks/1/path = NodePath("Highlight/PointLight2D:energy")
+tracks/1/interp = 1
+tracks/1/loop_wrap = true
+tracks/1/keys = {
+"times": PackedFloat32Array(0.5, 2.1, 4.4, 5),
+"transitions": PackedFloat32Array(1, 2, 1, 2),
+"update": 0,
+"values": [0.0, 0.8, 0.8, 0.0]
+}
+
+[sub_resource type="AnimationLibrary" id="AnimationLibrary_xveqw"]
```

Comment:
As this is now a separate scene, it is pretty questionable how `CreatureStageHUDBase.OnEditorButtonMouseEnter` works. So I think that script functionality should be split into a separate file and class called EditoryEntryButton.cs.

Reply:
That does stuff like this:
```
        editorButton.GetNode<TextureRect>("Highlight").Hide();
        editorButton.GetNode<AnimationPlayer>("AnimationPlayer").Stop();
```

So it explicitly refers to the child nodes of a separate scene. That's a big problem in terms of encapsulation, reuse and decoupling. This needs to be redone with a EditorEntryButton class that handles modifying its internal child nodes.

Reply:
What do you mean? I'm saying you should remove the mouse enter and exit signals from the separate HUD scenes and put them in the new editor entry scene, then write a C# class that implements that functionality for that scene.

Reply:
That was the only immediate thing I saw. If the root node of the new scene is a button, then letting the button click registration and enable / disable work like they currently do is perfectly fine to me.